src/main/java/bio/terra/app/controller/SearchApiController.java

```diff
@@ -109,11 +108,10 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
         List<UUID> accessibleIds =
             iamService.listAuthorizedResources(getAuthenticatedInfo(), IamResourceType.DATASNAPSHOT);
 
-        final List<String> snapshotIds = searchQueryRequest.getSnapshotIds();
+        final List<UUID> snapshotIds = searchQueryRequest.getSnapshotIds();
 
         Set<UUID> requestIds =
-            snapshotIds == null || snapshotIds.isEmpty() ? Collections.emptySet() :
-                snapshotIds.stream().map(UUID::fromString).collect(Collectors.toSet());
+            snapshotIds == null || snapshotIds.isEmpty() ? Collections.emptySet() : new HashSet<>(snapshotIds);
```

Comment:
Minor improvement to use the new unmodifiable `Set` factory methods
```suggestion
            snapshotIds == null || snapshotIds.isEmpty() ? Set.of() : Set.copyOf(snapshotIds);
```